willow/config/routes.rb

```diff
@@ -32,3 +32,6 @@
 
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
 end
+
+# Used to generate URLs in libraries that do not have full access to the Rails stack. Will default to the email host name if not provided.
+Rails.application.routes.default_url_options[:host] = ENV['DEFAULT_URL_OPTIONS_HOST'] || ENV['EMAIL_BASE_URL_HOST'] || 'localhost'
```

Comment:
This is a bit concerning, since we have something very similar in the ActionMailer initialiser. I see that this is for libs that don't have access to all of Rails (not sure exactly what that means but I can sort of guess). All fine so far.

I think it's worth trying to get the ActionMailer initialiser to use the `Rails.application.routes.default_url_options` being introduced. This will avoid configuration and Env var duplication of how we represent "the base URL in a non-web context like email or async API message sending".

Reply:
This is the ActionMailer initialiser : https://github.com/CottageLabs/willow/blob/master/willow/config/initializers/action_mailer.rb . As you can see, the setting is even called the same name, `default_url_options`.